config.go

```diff
@@ -83,20 +84,25 @@ type Config struct {
 
 	LoggerFactory logging.LoggerFactory
 
-	// ConnectTimeout is the timeout threshold for new connection handshakes
-	// to complete (default is 30 seconds)
-	ConnectTimeout *time.Duration
+	// function to make a context used in Dial(), Client(), Server(), and Accept().
+	// If nil, DefaultConnectContextMaker is used.
+	ConnectContextMaker func() (context.Context, func())
```

Comment:
This function type suggest user to use cancelable contexts.

Reply:
I am fine removing this entirely! ping @daenney make sure I am not making a bad decision :)

* `Context` with a deadline if they still need deadlines.
* We don't support Dialer, so not like we are breaking any APIs (Dialer also doesn't support server so will not work for us)

Reply:
I added this to make it possible to customize the behavior of accepted connection in `dtls.listener`.

As `Listen()` blocks until incoming connection and return established `dtls.Conn`, deadline timer must be initialized at the time the first packet, that triggers dtls handshake, arrives.

Reply:
Thanks for the ping. I think this makes sense, and the fact that when not explicitly provided we inject a default context with a proper timeout makes me happy.